pkg/resource/db_cluster/custom_update.go

```diff
@@ -561,7 +562,12 @@ func (rm *resourceManager) newCustomUpdateRequestPayload(
 		res.SetEnableIAMDatabaseAuthentication(*desired.ko.Spec.EnableIAMDatabaseAuthentication)
 	}
 	if desired.ko.Spec.EngineVersion != nil && delta.DifferentAt("Spec.EngineVersion") {
-		res.SetEngineVersion(*desired.ko.Spec.EngineVersion)
+		r := regexp.MustCompile(`[0-9]*$`)
+		desiredMajorEngineVersion := r.ReplaceAllString(*desired.ko.Spec.EngineVersion, "${1}")
```

Comment:
nil, we need to handle the case of nil `*desired.ko.Spec.EngineVersion`

Reply:
Thanks a lot for you review @a-hilaly. If `*desired.ko.Spec.EngineVersion` is nil it'll not perform any update, isn't it the expected behavior?